core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/commands/ConcretizeFunctionAllocationCommand.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2023 THALES GLOBAL SERVICES.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *    Thales Global Services - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.platform.sirius.ui.commands;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.emf.common.util.EList;
+import org.polarsys.capella.common.data.modellingcore.ModelElement;
+import org.polarsys.capella.core.data.cs.Component;
+import org.polarsys.capella.core.data.ctx.SystemFunction;
+import org.polarsys.capella.core.data.fa.AbstractFunction;
+import org.polarsys.capella.core.data.fa.AbstractFunctionalBlock;
+import org.polarsys.capella.core.data.fa.ComponentFunctionalAllocation;
+import org.polarsys.capella.core.data.fa.FaFactory;
+import org.polarsys.capella.core.data.helpers.fa.services.FunctionExt;
+import org.polarsys.capella.core.data.la.LogicalFunction;
+import org.polarsys.capella.core.data.pa.PhysicalFunction;
+import org.polarsys.capella.core.model.helpers.AbstractFunctionExt;
+
+/**
+ * Command for the ConcretizeFunctionAllocation accelerator
+ * 
+ * @author ebausson
+ */
+public class ConcretizeFunctionAllocationCommand extends AbstractFixCommand {
+
+  public ConcretizeFunctionAllocationCommand(Collection<ModelElement> selection) {
+    super(selection);
+  }
+
+  @Override
+  protected void process(ModelElement element) {
+    if ((element instanceof LogicalFunction) || (element instanceof SystemFunction) || (element instanceof PhysicalFunction)) {
+      AbstractFunction motherFunction = (AbstractFunction) element;
+      Component component = resolveMotherFunctionAllocation(motherFunction);
+      if (component != null) {
+        createAlloction(component, motherFunction);

```

Comment:
functionally, doesn't seems arcadia legit to allocate the mother function. only leafs can be allocated to components. to be discussed. Is there a DOREMI related ?
![Uploading image.png…]()

Reply:
Would it be acceptable if it was a config option (that was deactivated by default)?

Reply:
Hi, 
It needs to be discussed. 
Is there a DOREMI related describing the need to be discussed while obeo sync meeting?

Reply:
No, this is not DOREMI related, it is a client request.
Woulda PR with the functionality deactivated by default (and activable through the preferences) have better chance to be merged?